File: java-boot/src/test/java/se/callistaenterprise/kotlindsl/data/OrderDslTest.java
package se.callistaenterprise.kotlindsl.data;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.testcontainers.shaded.com.google.common.io.Resources;
import se.callistaenterprise.kotlindsl.testutil.DslIntegrationTest;
import se.callistaenterprise.kotlindsl.testutil.DslSpringBootTest;

import java.io.IOException;
import java.net.URL;
import java.nio.charset.StandardCharsets;

@DslSpringBootTest
public class OrderDslTest extends DslIntegrationTest {

    private final DataRepositories repos;

    public OrderDslTest(@Autowired DataRepositories repos) {
        this.repos = repos;
    }

    @Test
    public void testDemodataFile() throws IOException {
        var dsl = DslUtil.evalScript(fileContents("dsl/demodata.orderdsl.kts"), "demodata.orderdsl.kts");
        repos.persist(dsl);
    }

    private String fileContents(String file) throws IOException {
        URL url = Resources.getResource(file);
        return Resources.toString(url, StandardCharsets.UTF_8);
    }

}

File: kotlin-boot/src/test/kotlin/se/callistaenterprise/kotlindsl/data/OrderDslTest.kt
package se.callistaenterprise.kotlindsl.data

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import se.callistaenterprise.kotlindsl.testutil.DslIntegrationTest
import se.callistaenterprise.kotlindsl.testutil.DslSpringBootTest

@DslSpringBootTest
class OrderDslTest(@Autowired val repos: DataRepositories) : DslIntegrationTest() {

    @Test
    fun `Test inline`() {
        repos.persist {
            val ann = customer {
                firstName = "Ann"
                lastName = "Svensson"
            }

            val cal = customer {
                firstName = "Cal"
                lastName = "Carlsson"
            }

            val computer = product("Computer")
            val keyboard = product("Keyboard")
            val mouse = product("Mouse")

            order(ann) {
                item(computer, 2)
                item(keyboard)
                item(mouse)
            }

            order(cal) {
                item(computer)
                item(keyboard)
                item(mouse, 2)
            }
        }
    }

    @Test
    fun `Test demodata file`() {
        val dsl = DslUtil.evalScript(fileContents("/dsl/demodata.orderdsl.kts"), "demodata.orderdsl.kts")

        repos.persist(dsl)
    }

    private fun fileContents(file: String) = this::class.java.getResource(file)!!.readText()

}
